pilot/pkg/model/context.go

```diff
@@ -533,10 +537,11 @@ type NodeMetadata struct {
 	// Mostly used when istiod requests the upstream.
 	IstioRevision string `json:"ISTIO_REVISION,omitempty"`
 
-	// Labels specifies the set of workload instance (ex: k8s pod) labels associated with this node.
+	// Labels specifies the set of workload instance (ex: k8s pod) labels associated with this node. Labels is a
+	// superset of IstioMetaLabels.
 	Labels map[string]string `json:"LABELS,omitempty"`
```

Comment:
IMO this feels a bit backwards. We have `Proxy{MetaLabels: map, NodeMeta: {Labels: map}`.

Shouldn't it be:

`Proxy{Labels: map, NodeMeta: {MetaLabels: map}`? Its confusing that compute labels are in "node metadata" but the labels from metadata are not?

Reply:
The reason is we largely depend on NodeMeta.Labels across the repo, should change in a separate pr

Reply:
Ok, maybe we can merge that PR first and then do this. It seems very confusing to have it inverted

Reply:
sure, i will work on refactor first